sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosClientCache.scala

```diff
@@ -275,6 +275,19 @@ private[spark] object CosmosClientCache extends BasicLoggingTrait {
                   .setIoThreadPriority(directConfig, Thread.MAX_PRIORITY)
 
           builder = builder.directMode(directConfig)
+
+          if (cosmosClientConfiguration.proactiveConnectionInitialization.isDefined &&
```

Comment:
since we exposed the config, then we can clean up the code in SparkUtils.safeOpenConnectionInitCaches